internal/common/test_configuration.go

```diff
@@ -97,6 +98,61 @@ func GenerateTestDefaultInventoryIPv4Only() string {
 	Expect(err).To(Not(HaveOccurred()))
 	return string(b)
 }
+func GenerateTestDefaultInventoryWithIPv4(ipv4, hostname string) string {
+	inventory := &models.Inventory{
+		CPU: &models.CPU{Count: 8},
+		Disks: []*models.Disk{
+			{
+				SizeBytes: conversions.GibToBytes(120),
+				DriveType: "HDD",
+			},
+		},
+		Interfaces: []*models.Interface{
+			{
+				Name: "eth0",
+				IPV4Addresses: []string{
+					ipv4,
+				},
+			},
+		},
+		Memory:       &models.Memory{PhysicalBytes: conversions.GibToBytes(16), UsableBytes: conversions.GibToBytes(16)},
+		Hostname:     hostname,
+		SystemVendor: &models.SystemVendor{Manufacturer: "Red Hat", ProductName: "RHEL", SerialNumber: "3534"},
+		Timestamp:    1601835002,
+	}
+
+	b, err := json.Marshal(inventory)
+	Expect(err).To(Not(HaveOccurred()))
+	return string(b)
+}
+
+func GenerateTestDefaultInventoryWithIPv6(ipv6, hostname string) string {
```

Comment:
This function and `GenerateTestDefaultInventoryWithIPv4` are almost identical; please extract common code to a separate function.

Reply:
Understood.